### DISCUSSION TITLE
Best (general-purpose) practice for cutting off gene ontology annotation

### DISCUSSION THREAD

**Discussion by jasperhyp** [geneontology/helpdesk#419] [Q&A] [answered]
Hi, I've collected annotations of a large number of genes from GO, and propagated annotations back to the GO ancestors. Now I am trying to use these annotations for a general-purpose genomic functional graph (similar to genomic KG, etc. that are recently developed). The problem is that those top-level terms (e.g., on levels 0 and 1) really don't make sense. However, I don't really know at which level I should be cutting off the GO terms... And there are also papers cutting off terms based on the number of genes associated with such terms.

Can someone recommend some papers or approaches that contain details on essentially trimming GO terms w.r.t GO level or #genes associated (whether before or after propagation of GOAs), etc.?

**Comment by hattrill** [comment]
I would suggest you use one of the GO subsets available http://geneontology.org/docs/download-ontology/

The 'Generic GO subset' is one that should be broadly useful for your case, giving enough granularity to map genes to meanful classes. Organism-specific subsets are also available.

**Reply by jasperhyp** [reply]
Thank you!  This is very interesting, I'll experiment with this approach.  Where can I find some documents as to how this specific subset was chosen?  Is whether or not a GO term is kept based on, e.g., number of genes annotated with such GO term?

**Comment by hattrill** [comment, marked as answer]
Hi @jasperhyp we haven't written this up yet. The generic subset was updated fairly recently via a GO consortium working group and with feedback from GOC members. 

This subset is directed by grouping in biological classes that are physiologically relevant - so at a level that is not too specific and not too general. So, gene numbers/class are not important per se, but might be used to see if a class is too broad or too narrow. The major emphais was on coverage and - getting the number of genes with annotations that do not fall into the subset (across species) as low as possible. Also, producing classes that are not redundant/overlapping was also considered.

PomBase have some useful doc on GO slims: https://www.pombase.org/browse-curation/fission-yeast-go-slimming-tips which captures this essence of the process we followed. Their paper: https://royalsocietypublishing.org/doi/10.1098/rsob.180241 also captures the major aims of the generic slim that we followed.

**Reply by jasperhyp** [reply]
Thank you very much for the detailed explanation!  I really appreciate it.  Two final questions:

1. When mapping GOA (e.g. human GOA from EBI) to this subset of GO terms, do we just duplicate the annotations on GO terms that are missing in the subset to their most recent ancestors in the subset?

2. I noticed that some GO terms in the subset do not have any relationships, including `is_a`.  I suspect this is due to the removal of its parents. What are some common practices you would suggest to put them back into the ontology?  I am thinking of keeping all GO terms (from `go-basic.obo`) for calculating ontology-based properties (e.g. MRCA, random walk, etc.), but separately tracking whether a term is in the "meaningful" subset (from `goslim_generic.obo`) or not.

**Reply by hattrill** [reply]
I think what you are asking can be found by using the  [go-basic.obo](http://geneontology.org/docs/download-ontology/#go_basic)  The subsets are 'labelled' in the obo file using a subset tag: subset: goslim_generic (subsets are listed in the header of the GO file), so they are in whole the GO heirarchy.

The tool below may also help you achieve this in an easier way:
I very much like the tool at https://go.princeton.edu/ for looking at mapping to subsets. You can alsio visualize them in a tree view. 
The tool at QuickGO is also good for looking at slimmed sets: https://www.ebi.ac.uk/QuickGO/slimming

**Reply by jasperhyp** [reply]
Thank you so much!
